.github/workflows/test_sign_xcframework.yml

```diff
@@ -0,0 +1,46 @@
+name: Test Build Signed XCFramework
+run-name: Test Build Signed XCFramework
+
+on:
+  pull_request:
+
+jobs:
+  prepare_framework:
+    runs-on: macos-13-xlarge
```

Comment:
`h`: Please don't use an xlarge runner. Running this on every PR costs easily around 2k $ a month or most likely more. We obviously can spend 2k a month, but only if it makes sense. Let's use a `macos-14` please. We can't use macos-13 as mentioned here
https://github.com/getsentry/sentry-cocoa/blob/aa48954db865ed82e38feb313885dc6cb372cc8b/.github/workflows/build.yml#L101-L105

Reply:
@itaybre you can convert it back to draft